bindings/python/src/modelling/expose-force-cost.cpp

```diff
@@ -51,7 +51,7 @@ void exposeContactForce() {
       .def(bp::init<int, PinModel, const context::MatrixXs &,
                     const RigidConstraintModelVector &,
                     const pinocchio::ProximalSettingsTpl<Scalar> &,
-                    const context::Vector6s &, const std::string &>(bp::args(
+                    const context::VectorXs &, const std::string &>(bp::args(
```

Comment:
I don't understand this change

Reply:
In this context, the reference force can be a 3D or 6D vector, depending on the type of contact. The bindings were not covering the 3D case.

Reply:
Okay, so in this case, since at most it can be of dimension six, you could use the `Eigen::Matrix` template with maximum number of rows at compile time set to 6, in both the binding code and the actual C++ class:

```cpp
using Vector3or6 = Eigen::Matrix<Scalar, -1, 1, 6>;
```

The fourth template parameter is known in Eigen as `MaxRowsAtCompileTime`.
This allows us to avoid an unnecessary heap allocation in this case.